docs/design/jet/025-jet-job-partition-pruning.md

```diff
@@ -123,9 +123,85 @@ try to choose only the required members.
+Each inner list of expressions contains comparison expressions, and may be **one-sized**, 
+if the partitioning strategy key is **simple**, and **multi-sized**, if the partitioning strategy key is **composite**. 
+If we have more than one prunable filter predicate, outer list will be multi-sized. 
+
+**Currently, it is implemented only for IMap connector, where all expressions are available.**
+
+For better imagination we prepared an example below.
+
+#### Successful case example 
+Let's assume we have an IMap `map` with composite key {comp1, comp2, comp3} and applied partitioning strategy for
+`comp1` and `comp2`. Let's have the following synthetic query, where filter matches the partitioning strategy:
+```
+SELECT * FROM map WHERE __key.comp1 = 1 AND __key.comp2 = 2
+```
+IMap-specific `fullScanReader` receives the following list of maps as a parameter:
+```
+[{"__key.comp1" = Expression(`__key.comp1 = 1`)}, {"__key.comp2" = Expression(`__key.comp2 = 2`)}]
+```
+
+After the described computation above, `fullScanReader` implementation should return the following list of expressions:
+```
+[[Expression(`__key.comp1 = 2`], [Expression(`__key.comp1 = 2`]]
+```
+
+#### Not successful case example
+
+For same `map` as above, let's have the following synthetic query, where filter does not match the partitioning strategy:
+```
+SELECT * FROM map WHERE __key.comp1 = 1 AND __key.comp3 = 3
+```
+
+IMap-specific `fullScanReader` receives the following list of maps as a parameter:
+```
+[{"__key.comp1" = Expression(`__key.comp1 = 1`)}, {"__key.comp2" = Expression(`__key.comp3 = 3)}]
+```
+
+but, instead of example below, it will return an empty list, because partition pruning is not applicable due to presence
+of `__key.comp3` in the filter, which was not declared as part of partitioning strategy.
+
+**Important note : simple `__key` is always participates in default partitioning strategy, since IMap is partitioned
+by __key**.
+
+#### IMap-specific prunable meta supplier implementation details
+
+`SpecificPartitionsImapReaderPms` is a new meta supplier to perform IMap scans. It was designed with ability 
+to self-sufficiently calculate exact set of partitions to scan, similarly to `lazyForceTotalParallelismOne` 
+described below. As a parameter, it accepts an optional list of lists of expressions (see previous section), 
+which triggers the code path for partition pruning. Also, `SpecificPartitionsImapReaderPms` has a possibility not to do
+any partitions' calculation, if partitioning strategy is not available for the given IMap.
+
+Regarding partition pruning case, the related calculation is performed by evaluating received list of expressions in 
+`init` method, where partition assignment is available, and as a result, we receive available partition set. Later,
+in `get` method we compute required partitions subset per member (stored in `ProcessorSupplier`), and then each
```

Comment:
you skipped important detail (see above comment): will the PMS be reusable? I could deduce that from english code but the point of TDD is to describe such descisions/requirements/assumptions explicitly.

Reply:
Argh, I am always missing this new 'feature'